packages/flutter/lib/src/material/data_table.dart

```diff
@@ -42,6 +42,7 @@ class DataColumn {
     this.numeric = false,
     this.onSort,
     this.mouseCursor,
+    this.mainAxisAlignment,
```

Comment:
Oh gosh I feel silly for not noticing this earlier. This should probably be crossAxisAlignment, technically. We are talking about the horizontal alignment of DataColumn, for which the main axis would technically be vertical. 

Or we could call it headerMainAxisAlignment, since it only applies to the header rather than the whole column. WDYT?

Reply:
> Oh gosh I feel silly for not noticing this earlier. This should probably be crossAxisAlignment, technically. We are talking about the horizontal alignment of DataColumn, for which the main axis would technically be vertical.

Great observation! I was focused on `Row` widget inside the `DataColumn`. I didn't consider what it means for the `DataColumn` with column in the name.

>  Or we could call it headerMainAxisAlignment, since it only applies to the header rather than the whole column. WDYT?

Sounds good to me. This could be consistent with other header properties if we do `headingRowAlignment`

https://github.com/flutter/flutter/blob/859a5ab0f5a918f0d81b62286e519367662f12a4/packages/flutter/lib/src/material/data_table_theme.dart#L51-L53

I also added this to `DataTableTheme`.

Reply:
Perfect!